client/dashboard/sites/site-delete-modal/index.tsx

```diff
@@ -0,0 +1,247 @@
+import { DataForm } from '@automattic/dataviews';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useRouter } from '@tanstack/react-router';
+import {
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Button,
+	ExternalLink,
+	Modal,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { useAuth } from '../../app/auth';
+import {
+	deleteSiteMutation,
+	p2HubP2sQuery,
+	siteHasPurchasesCancelableQuery,
+} from '../../app/queries';
+import Notice from '../../components/notice';
+import type { Site } from '../../data/types';
+import type { Field } from '@automattic/dataviews';
+
+type SiteDeleteFormData = {
+	domain: string;
+};
+
+const canDeleteSite = ( site: Site ) =>
+	( site.is_wpcom_atomic || ! site.jetpack ) && ! site.is_vip && ! site.options?.p2_hub_blog_id;
+
+const TRIAL_PRODUCT_SLUGS = [
+	'wp_bundle_migration_trial_monthly',
+	'wp_bundle_hosting_trial_monthly',
+	'ecommerce-trial-bundle-monthly',
+];
+
+export function SiteDeleteWarningModal( { site, onClose }: { site: Site; onClose: () => void } ) {
+	const { data: p2HubP2s } = useQuery( {
+		...p2HubP2sQuery( site.slug, { limit: 1 } ),
+		enabled: !! site.options?.p2_hub_blog_id && site.options?.is_wpforteams_site,
+	} );
+
+	const isAtomicRemovalInProgress = site.plan?.is_free && site.is_wpcom_atomic;
+	const p2HubP2Count = p2HubP2s?.totalItems ?? 0;
+	const isTrialSite =
```

Comment:
suggestion: consider moving this logic and the TRIAL_PRODUCT_SLUGS into a single function since it's fairly self-contained.